Ps_recog[] ps_recog
uint16 ps_numb